metadata.go

```diff
@@ -80,12 +82,36 @@ var debianLicenseText = map[string]string{
 `,
 }
 
+var githubRegexp = regexp.MustCompile(`github\.com/([\w._-]+/[\w._-]+)`)
+
+// regex match github.com/([^/]+/[^/]+) on reply of https://gopkg/?go-get=1
+func findGitHubRepo(gopkg string) (string, error) {
+	if strings.HasPrefix(gopkg, "github.com/") {
+		return gopkg[len("github.com/"):], nil
+	}
+	resp, err := http.Get("https://" + gopkg + "?go-get=1")
+	if err != nil {
+		return "", err
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return "", err
+	}
+	repoMatch := githubRegexp.FindSubmatch(body)
```

Comment:
regexp indeed is simpler, but my concern is not about simplicity, but reliability. It’s probably fine to start with the regexp approach, but please add a TODO to actually parse go-import and go-source.